File: PyLabFastAPI/app/schemas/media.py
# PyLabFastAPI/app/schemas/media.py
from pydantic import BaseModel

class VideoCreateReq(BaseModel):
    title: str          # 视频标题
    file_key: str       # 七牛云返回的文件 Key
    file_hash: str      # 文件指纹 (etag)
    file_size: int      # 文件大小
    duration: int = 0   # 视频时长 (秒)

File: PyLabVUE/src/model/media.ts
// src/model/media.ts

/**
 * 视频创建参数
 */
export interface CreateVideoReq {
  title: string
  file_key: string // 七牛云的文件 Key
  file_hash: string // 七牛云的文件 Hash
  file_size: number
  duration?: number // 视频时长 (可选)
}

/**
 * 视频信息响应
 */
export interface VideoInfo {
  id: number
  title: string
  play_url: string
  cover_url: string
  created_at: string
}
